op-challenger/game/keccak/matrix/matrix_test.go

```diff
@@ -220,6 +195,126 @@ func TestMatrix_AbsorbNextLeaf(t *testing.T) {
 	}
 }
 
+func TestVerifyPreimage(t *testing.T) {
+	preimage := testutils.RandomData(rand.New(rand.NewSource(2323)), 1024)
+	validCommitments := func() []common.Hash {
+		s := NewStateMatrix()
+		valid, err := s.AbsorbUpTo(bytes.NewReader(preimage), 1000*types.BlockSize)
+		require.ErrorIs(t, err, io.EOF, "Should read all preimage data")
+		return valid.Commitments
+	}
+	leafData := func(idx int) [types.BlockSize]byte {
+		var out [types.BlockSize]byte
+		copy(out[:], preimage[idx*types.BlockSize:(idx+1)*types.BlockSize])
+		return out
+	}
+	challengeLeaf := func(commitments []common.Hash, invalidIdx int) *types.Challenge {
+		invalidLeafStart := invalidIdx * types.BlockSize
+		s := NewStateMatrix()
+		_, err := s.AbsorbUpTo(bytes.NewReader(preimage), invalidLeafStart)
+		require.NoError(t, err)
+
+		return &types.Challenge{
+			StateMatrix: s.PackState(),
+			Prestate: types.Leaf{
+				Input:           leafData(invalidIdx - 1),
+				Index:           big.NewInt(int64(invalidIdx - 1)),
+				StateCommitment: commitments[invalidIdx-1],
+			},
+			//PrestateProof: nil,
```

Comment:
Should there be `TODO` comments here for follow up after the merkle tree implementation?

Reply:
eh. I probably should have just removed the commented out code - updating the test is a natural part of doing the work to add the merkle proofs.  Actually I am going to remove those comments and the TODO about it in the Challenge type declaration (which I phrased wrong anyway). We have the issue still open as the TODO so really only need comments in the code when something is potentially dangerous if missed (eg confusing padded and unpadded values) or returning a not implemented error (even that's probably borderline).  Where we just don't support some functionality the issue should be enough.